theano/tensor/basic.py

```diff
@@ -5128,6 +5134,8 @@ class Diagonal(Op):
     """
 
     def __init__(self, offset=0, axis1=0, axis2=1):
+        if numpy_diagonal_return_view:
+            self.view_map = {0: [0]}
```

Comment:
I thought we were not supposed to have views before a certain point of the optimization in general?